rero/rero-ils
# Paging (stack request) functionnalities

This commit implements tasks relative to UCL paging request in closed stack.
It allows to limit request for items belonging to specific location.
It also allows to restrict the pickup destination to some pickup locations
instead of all pickups locations of the organisation.
If an item of a "paging" location is requested, it also allows to send
a notification to notify a manager.



## how to test

Nee to test with RERO-ILS-UI : https://github.com/rero/rero-ils-ui/pull/171

* logged as a librarian, create a new location into your library. First step is to set the flag "allow_request" to false.
* create a new item an link it to this new location
* On an other browser, logged as a patron and try to request this item. The "request" button shouldn't be displayed
----
* on your first browser (logged as librarian) edit the new location : set "allow request" to true, and restrict pickup location to one or more (not all) possible pickups locations.
* on the second browser (logged as patron) refresh your page and try to request the item. The request select bow should only contains pickups locations that you select as possible pickups locations.
----
* on librarian browser : edit the location and set "send_notification" to true and set an email address into the "contact_email" field.
* on the patron browser : place a request on the newly created item. In your RERO-ILS console your should see than an email is send to location email address.
----
* Into the professional interface, all "request" button/select should follow the same behavior.

## Code review check list
